spec/exe/metasploit_cache_spec.rb

```diff
@@ -109,6 +109,66 @@ def db_schema_load
           # Context Methods
           #
 
+          def self.reference_name_context(module_type, reference_name, pending_reason_by_reference_name: {}, run_counter:)
+            context_options = {}
+            pending_reason = pending_reason_by_reference_name[reference_name]
+
+            if pending_reason
+              context_options[:pending] = pending_reason
+            end
+
+            context reference_name, context_options do
+              unless pending_reason
+                if run_counter.max
+                  before(:each) do
+                    run_counter.count += 1
+
+                    if run_counter.count > run_counter.max
+                      skip "Skipping because #{run_counter.max} Metasploit Modules have been tested already and testing them all takes too long"
+                    end
+                  end
+                end
+              end
+
+              full_name = "#{module_type}/#{reference_name}"
+
+              it "can be `use`d" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.